Enable dual-stack endpoint support for ec2 describe-instances during AWS discovery

tanmay17061 opened the pull request:
Current AWS discovery logic fails in IPv6-only environments as `ec2 describe-instances` defaults to accessing AWS's IPv4 endpoints. This change enables dual-stack support by default, allowing the use of IPv6 endpoints where available.

Tested working in:

- [x] an IPv6-only env

- [x] an IPv4-only env

abhijeetviswa commented on provider/aws/aws_discover.go:
```diff
@@ -149,6 +149,7 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 		l.Printf("[INFO] discover-aws: Using default credential chain")
 		cfg, err = config.LoadDefaultConfig(context.TODO(),
 			config.WithRegion(region), // Specify your region
+			config.WithUseDualStackEndpoint(aws.DualStackEndpointStateEnabled),
```
Question: Why is dual stack limited to only the default credential chain?

tanmay17061 replied:
thanks. you're right, I have raised another commit to enable dual-stack in non-default flow as well.

tanmay17061 commented:
hey @abhijeetviswa,
I've addressed your comment.
While I've tested on Consul, wanted to ask if this MR can have any regressions on other products. Is there any workflow in place to check this?

abhijeetviswa commented:
@tanmay17061 The changes look good to me. However, there are failing acceptance tests. Also, please sign the CLA bot. 

> Is there any workflow in place to check this?

I'm not aware of any workflow or process. @mukeshjc perhaps you can help?

tanmay17061 commented:
@abhijeetviswa
Thanks for your comment!
I can see that it requires you to resolve your previous review to re-run the CI pipeline.

abhijeetviswa commented:
> I can see that it requires you to resolve your previous review to re-run the CI pipeline.

I see it's been resolved. Do you need me to do anything else?

tanmay17061 commented:
Closing to re-open to re-trigger CI

tanmay17061 commented:
re-opening
